odoo/addons/base/models/qweb.py

```diff
@@ -254,9 +255,13 @@ def _render(self, template, values=None, **options):
             * ``profile`` (float) profile the rendering (use astor lib) (filter
               profile line with time ms >= profile)
         """
+        values = values or {}
         body = []
-        self.compile(template, options)(self, body.append, values or {})
-        return u''.join(body).encode('utf8')
+        self.compile(template, options)(self, body.append, values)
+        joined = u''.join(body)
+        if not values.get('__keep_empty_lines'):
+            joined = QWeb._RETURNS.sub('\n', joined.lstrip())
```

Comment:
while lstrip ? and not strip(). Do we really want trailing whitespace?

Reply:
In some situations, rendered templates are used as values inside other templates.
While we do not want trailing spaces, I believe we still want the final line feed, so that the caller template resumes on the next line.

Reply:
in this case, the left too ;) so probably no strip at this end

Reply:
Right. So it is the responsibility of the nesting-caller to decide whether to strip its content or not.
BTW, I also noticed that the mechanism I mentioned is actually using QWeb JS - so my argument was wrong.